pkg/controllers/provisioning/suite_test.go

```diff
@@ -145,6 +147,26 @@ var _ = Describe("Provisioning", func() {
 				ExpectScheduled(ctx, env.Client, pod)
 			}
 		})
+		Context("Instance Type Selection", func() {
+			BeforeEach(func() {
+				ctx = injection.WithTestInstance(ctx, true)
+			})
+
+			AfterEach(func() {
+				ctx = injection.WithTestInstance(ctx, false)
+			})
+
+			It("should not select invalid instance types", func() {
```

Comment:
I would explicitly use a specific instance type (using a node selector) and then verify that the pod can't be scheduled.

Reply:
So here is the minor catch. The illy created instance types that I add will also fails the other tests in all the suites. Unless we also specify node selector for other tests to exclude these illy created instance types, it may create confusions what the problem is.